feat : create component   top app bar

nouha06 opened the pull request:
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Related issues

- #91 
### Description

<!-- Describe your changes in detail -->

### Motivation & Context

<!-- Why is this change required? What problem does it solve? -->

### Types of change

<!-- What types of changes do your code introduce? -->
<!-- Please remove the unused items in the list -->

- Bug fix (non-breaking which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Refactoring (non-breaking change)
- Breaking change (fix or feature that would change existing functionality)

### Previews

<!-- Please add screenshots or videos showing your evolutions -->

### Checklist

- [x] Implement App bar for Android devices
- [x] Implement navigation bar for iOS devices
- [x] Manage the a11y feature

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!-- Note that any checkboxes not optional must be ticked by an 'x' (or '(NA)') and our [branch ruleset](https://github.com/marketplace/task-list-completed) may block any merge if some mandatory boxes remain empty -->
<!-- Your branch used to submit the evolutions must be prefixed by the issue number like 666-add-some-feature -->

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/ouds-flutter/blob/develop/CONTRIBUTING.md)

#### Accessibility

- [x] My change follows accessibility good practices

#### Design

- [x] My change respects the design guidelines of _Orange Unified Design System_

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/ouds-flutter/blob/develop/DEVELOP.md)
- [ ] I have added unit tests to cover my changes _(optional)_

#### Documentation

- [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- [x] Manually test (dark mode, RTL, landscape display, tablet)
- [x] Documentation has been updated if relevant
- [x] Design review
- [ ] A11y review
- [x] Internal files have been updated if relevant (THIRD_PARTY, NOTICE)
- [x] changelog.md has been updated respecting [keep a changelog rules](https://keepachangelog.com/en/1.0.0/) and referencing the issue

nouha06 committed:
chore: add ouds app bar component first commit

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

nouha06 committed:
chore : add medium and large top app bar

nouha06 committed:
chore : add badge component in oudsButton
add code generator for top appbar and finishing top app bar in demo app for android devices

nouha06 committed:
chore : add CupertinoNavigationBar in core for iOS

nouha06 committed:
chore : add CupertinoNavigationBar in demo app for iOS

nouha06 committed:
feat : Add rules in each type of device iPhone or iPad in lib and demo app

nouha06 committed:
feat : refactor top app bar in lib and update code generator in demo app

nouha06 committed:
feat : manage accessibility in top app bar component

nouha06 committed:
feat : manage accessibility in Cupertino navigation bar component

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

nouha06 committed:
feat : delete cupertino navigation bar

nouha06 committed:
chore : delete tag in component version

nouha06 committed:
feat: add expandedHeight and lineCount parameters in SliverAppBar

nouha06 committed:
doc: update documentation

nouha06 committed:
refactor: refactor code and add an ouds_avatar component for top appbar

AhmedAmineZr reviewed (commented):
il manque aussi la changeLog

nouha06 committed:
doc(navigation-bar): update doc

nouha06 committed:
feat(top-appBar): refactor icon badge in top app bar

nouha06 committed:
feat(top-appBar): refactor medium and large top app bar

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

nouha06 committed:
feat(top-appbar): update code generator and doc

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

# Conflicts:
#	app/lib/l10n/gen/ouds_flutter_app_localizations.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_ar.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_en.dart
#	app/lib/l10n/ouds_flutter_ar.arb
#	app/lib/l10n/ouds_flutter_en.arb
#	app/lib/ui/utilities/reference_design_version_component.dart

nouha06 committed:
feat(top-appbar): update doc

nouha06 committed:
feat(top-appbar): add OudsTopAppBarActionBadge, add close image for top app bar, add bottom border and blur effect

nouha06 committed:
feat(top-appbar): eliminate border

nouha06 committed:
chore: update changelog file and doc

nouha06 committed:
review: refactor code

nouha06 committed:
review:
- refactor code
- add a methode to generate a list for actions
- update labels
- clean code
- add comments

nouha06 committed:
review: add a control to not set invalid value for height of app bar

nouha06 committed:
feat: update code generator

nouha06 committed:
chore: add comments

nouha06 committed:
review: refactor code

nouha06 committed:
review: update doc

nouha06 committed:
review: update doc in lib and code generator in demo app

nouha06 committed:
feat: use the OudsTopAppBar in demo app

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

# Conflicts:
#	ouds_core/CHANGELOG.md
#	ouds_core/lib/l10n/gen/ouds_localizations.dart
#	ouds_core/lib/l10n/gen/ouds_localizations_ar.dart
#	ouds_core/lib/l10n/gen/ouds_localizations_en.dart
#	ouds_core/lib/l10n/ouds_flutter_ar.arb
#	ouds_core/lib/l10n/ouds_flutter_en.arb

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

nouha06 committed:
review: fix back and close icon button, fix monogram text and background color, use chips instead of text fields to select the maxLines"

nouha06 committed:
review: add boolean value to show avatar in trailing actions

AhmedAmineZr commented:
@nouha06 do not forget to update branch

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

# Conflicts:
#	pubspec.lock

nouha06 committed:
review: refactor code in demo app

nouha06 committed:
review: Add methode to calculate expanded height based on size and max lines

nouha06 committed:
feat: create component Toolbar Top

nouha06 committed:
feat: use TopBar in demo app

nouha06 committed:
review: update keys of labels

nouha06 committed:
Merge branch 'refs/heads/develop' into 582-create-component---tool-bar-top

# Conflicts:
#	app/CHANGELOG.md
#	app/lib/l10n/gen/ouds_flutter_app_localizations.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_ar.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_en.dart
#	app/lib/l10n/ouds_flutter_ar.arb
#	app/lib/l10n/ouds_flutter_en.arb
#	app/lib/ui/about/about_screen.dart
#	app/lib/ui/components/components.dart
#	app/lib/ui/components/components_screen.dart
#	app/lib/ui/main_screen.dart
#	app/lib/ui/tokens/token_screen.dart
#	ouds_core/CHANGELOG.md
#	ouds_core/lib/components/button/ouds_button.dart

nouha06 committed:
review: update app_components_common_action_a11y in arabic

nouha06 committed:
review: refactor code for ouds_top_bar_action_config.dart and modifiers

nouha06 committed:
Merge branch 'refs/heads/develop' into 582-create-component---tool-bar-top

# Conflicts:
#	app/lib/ui/components/pin_code_input/pin_code_input_demo_screen.dart

nouha06 committed:
review: refactor code and update doc in demo app

nouha06 committed:
review: refactor code and update doc

nouha06 committed:
Merge branch 'refs/heads/develop' into 582-create-component---tool-bar-top

# Conflicts:
#	app/lib/ui/components/badge/badge_demo_screen.dart
#	app/pubspec.yaml
#	ouds_core/CHANGELOG.md

nouha06 committed:
chore: use the LightDarkBox and update assets for orange compact theme

nouha06 committed:
Merge branch 'refs/heads/develop' into 91-create-component---top-app-bar

# Conflicts:
#	app/CHANGELOG.md
#	app/lib/l10n/gen/ouds_flutter_app_localizations.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_ar.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_en.dart
#	app/lib/l10n/ouds_flutter_ar.arb
#	app/lib/l10n/ouds_flutter_en.arb
#	app/lib/ui/about/about_screen.dart
#	app/lib/ui/about/detail/about_file_screen.dart
#	app/lib/ui/components/components.dart
#	app/lib/ui/components/components_screen.dart
#	app/lib/ui/main_screen.dart
#	app/lib/ui/tokens/token_screen.dart
#	app/pubspec.yaml
#	ouds_core/CHANGELOG.md
#	ouds_core/lib/components/button/ouds_button.dart

nouha06 committed:
merge: use LightDarkBox after rebase

nouha06 committed:
Merge branch 'refs/heads/91-create-component---top-app-bar' into 582-create-component---tool-bar-top

# Conflicts:
#	NOTICE.txt
#	app/CHANGELOG.md
#	app/lib/l10n/gen/ouds_flutter_app_localizations.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_ar.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_en.dart
#	app/lib/l10n/ouds_flutter_ar.arb
#	app/lib/l10n/ouds_flutter_en.arb
#	app/lib/main_app_bar.dart
#	app/lib/ui/about/material/material_component_screen.dart
#	app/lib/ui/components/badge/badge_demo_screen.dart
#	app/lib/ui/components/button/button_demo_screen.dart
#	app/lib/ui/components/checkbox/checkbox_demo_screen.dart
#	app/lib/ui/components/checkbox/checkbox_item_demo_screen.dart
#	app/lib/ui/components/chip/chip_filter_demo_sreen.dart
#	app/lib/ui/components/chip/chip_suggestion_demo_screen.dart
#	app/lib/ui/components/components.dart
#	app/lib/ui/components/components_screen.dart
#	app/lib/ui/components/divider/divider_demo_screen.dart
#	app/lib/ui/components/form_input/password_input/password_input_demo_screen.dart
#	app/lib/ui/components/form_input/phone_number/phone_number_input_demo_screen.dart
#	app/lib/ui/components/form_input/text_input/text_input_demo_screen.dart
#	app/lib/ui/components/pin_code_input/pin_code_input_demo_screen.dart
#	app/lib/ui/components/radio_button/radio_button_item_demo_screen.dart
#	app/lib/ui/components/switch/switch_demo_screen.dart
#	app/lib/ui/components/switch/switch_item_demo_screen.dart
#	app/lib/ui/tokens/border/border_screen.dart
#	app/lib/ui/tokens/color/color_screen.dart
#	app/lib/ui/tokens/elevation/elevation_screen.dart
#	app/lib/ui/tokens/opacity/opacity_screen.dart
#	app/lib/ui/tokens/token_screen.dart
#	app/lib/ui/tokens/typography/typography_screen.dart
#	app/lib/ui/utilities/app_assets.dart
#	app/lib/ui/utilities/customizable/customizable_textfield.dart
#	app/lib/ui/utilities/customizable/customizable_widget_state.dart
#	ouds_core/CHANGELOG.md
#	ouds_core/lib/components/avatar/ouds_avatar.dart
#	ouds_core/lib/components/button/internal/ouds_button_utils.dart
#	ouds_core/lib/components/button/ouds_button.dart
#	ouds_core/lib/l10n/gen/ouds_localizations.dart
#	ouds_core/lib/l10n/gen/ouds_localizations_ar.dart
#	ouds_core/lib/l10n/gen/ouds_localizations_en.dart
#	ouds_core/lib/l10n/ouds_flutter_ar.arb
#	ouds_core/lib/l10n/ouds_flutter_en.arb
#	ouds_theme_orange/pubspec.yaml
#	ouds_theme_sosh/pubspec.yaml
#	ouds_theme_wireframe/pubspec.yaml

nouha06 committed:
merge: fix merge conflict and update code generator

nouha06 committed:
Merge branch 'refs/heads/develop' into 582-create-component---tool-bar-top

# Conflicts:
#	app/CHANGELOG.md
#	app/ios/Podfile.lock
#	app/lib/main_app_bar.dart
#	ouds_core/CHANGELOG.md
#	ouds_core/lib/components/button/ouds_button.dart

nouha06 committed:
Merge branch 'refs/heads/develop' into 582-create-component---tool-bar-top

# Conflicts:
#	ouds_core/README.md

nouha06 committed:
fix: set padding and icon/label sizes to match native Figma navigation bar specifications

nouha06 committed:
chore: add fixed values for padding and size of icon in toolbar top and add the previous page title in lin and demo app

nouha06 committed:
Merge branch 'refs/heads/develop' into 582-create-component---tool-bar-top

nouha06 committed:
Merge branch 'refs/heads/develop' into 582-create-component---tool-bar-top

# Conflicts:
#	ouds_core/lib/components/button/ouds_button.dart

nouha06 committed:
chore: add previousTitle in code generator and status disabled for actions

nouha06 committed:
Merge branch 'develop' into 582-create-component---tool-bar-top

nouha06 committed:
review: add doc for previousPageTitle param

nouha06 committed:
chore: add semantic label for action button

nouha06 committed:
review: update changelogs and pubspec

nouha06 committed:
review: update the usage of previousPageTitle ind lib, doc and demo app

nouha06 committed:
review: refactor code

nouha06 committed:
Merge branch 'develop' into 582-create-component---tool-bar-top

nouha06 committed:
review: refactor code

nouha06 committed:
review: rename customIcon to icon

nouha06 committed:
Merge branch 'develop' into 582-create-component---tool-bar-top

# Conflicts:
#	app/lib/ui/components/badge/badge_demo_screen.dart
#	app/lib/ui/components/components.dart
#	app/lib/ui/components/tag/tag_demo_screen.dart

nouha06 committed:
chore: Update the color of the actions and implement the back label behavior following a long previous page title.

nouha06 committed:
Merge branch 'develop' into 582-create-component---tool-bar-top

nouha06 committed:
Merge branch 'develop' into 582-create-component---tool-bar-top

nouha06 committed:
chore: fix the flip back icon in rtl mode

nouha06 committed:
review: update padding and text size for actions

nouha06 committed:
fix: fix padding of icon action

nouha06 committed:
Merge branch 'develop' into 582-create-component---tool-bar-top

# Conflicts:
#	app/CHANGELOG.md
#	app/lib/ui/tokens/border/border_screen.dart
#	ouds_core/CHANGELOG.md

nouha06 committed:
feat: add french label and fic accessibility issue

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

# Conflicts:
#	app/CHANGELOG.md
#	app/lib/main_app_bar.dart
#	ouds_core/CHANGELOG.md
#	ouds_core/README.md
#	ouds_core/lib/components/button/ouds_button.dart

nouha06 committed:
merge: update labels

nouha06 committed:
Merge branch '91-create-component---top-app-bar' into 582-create-component---tool-bar-top

# Conflicts:
#	app/CHANGELOG.md
#	app/lib/l10n/gen/ouds_flutter_app_localizations_fr.dart
#	app/lib/main_app_bar.dart
#	ouds_core/CHANGELOG.md
#	ouds_core/lib/components/button/ouds_button.dart

nouha06 committed:
merge: update code generator

nouha06 committed:
Merge branch 'develop' into 582-create-component---tool-bar-top

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

nouha06 committed:
Merge branch '582-create-component---tool-bar-top' into 91-create-component---top-app-bar

nouha06 committed:
review: refactor use of previousPageTitle in demo app

nouha06 committed:
fix: bottom of the component screen is hidden by the bottom sheet

nouha06 committed:
fix: add comment in ConstantSheetBottom class

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

# Conflicts:
#	app/CHANGELOG.md
#	app/lib/ui/utilities/app_assets.dart

nouha06 committed:
fix: add missing comment on the top of ConstantSheetBottom class

nouha06 committed:
fix: update using of buttons in top bar

nouha06 committed:
review: update methode name

AhmedAmineZr committed:
feat:downgrade to sdk 3.35, remove warning & error

AhmedAmineZr committed:
Revert "feat:downgrade to sdk 3.35, remove warning & error"

This reverts commit 69aaf94b811c00f009c8d74f39c440d2fbec57b2.

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

nouha06 committed:
Merge branch 'develop' into 91-create-component---top-app-bar

# Conflicts:
#	app/CHANGELOG.md
